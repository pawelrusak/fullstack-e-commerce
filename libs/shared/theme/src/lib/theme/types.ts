export type * from './tokens/types';
export type * from './theme.types';
