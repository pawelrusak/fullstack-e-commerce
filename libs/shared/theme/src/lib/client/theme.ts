import { DefaultTheme } from 'styled-components';
import { colorPalette } from './color-palette';

const theme: DefaultTheme = {
  color: {
    primary: colorPalette.primary[500],
    onPrimary: 'hsl(0, 0%, 100%)',

    secondary: 'hsl(46, 99%, 65%)',

    surface: 'hsl(0, 0%, 100%)',
    onSurface: 'hsl(0, 0%, 19%);',
    onSurfaceVariant: colorPalette.primary[500],

    secondarySurface: 'hsl(0, 0%, 96%)',
    onSecondarySurface: 'hsl(0, 0%, 19%)',
    onSecondarySurfaceVariant: colorPalette.primary[500],

    background: 'hsl(0, 0%, 100%)',
    onBackground: 'hsl(0, 0%, 19%)',

    border: 'hsl(0, 0%, 96%)',
    error: 'hsl(348, 83%, 47%)',

    // TODO: Remove deprecated "secondBackground" color after refactoring
    secondBackground: 'hsl(0, 0%, 96%)',
    // TODO: Remove deprecated "text" color after replacing with "onBackground" or "onSurface"
    text: 'hsl(0, 0%, 19%)',
  },
  fontFamily: {
    // TODO: Remove deprecated "poppins" font family after refactoring
    poppins: ['Poppins', 'sans-serif'].join(', '),
    // TODO: Remove deprecated "montserrat" font family after refactoring
    montserrat: ['Montserrat', 'sans-serif'].join(', '),
    primary: ['Montserrat', 'sans-serif'].join(', '),
    secondary: ['Poppins', 'sans-serif'].join(', '),
  },
  fontWeight: {
    regular: 400,
    semiBold: 600,
    bold: 700,
  },
  space: [0, 0.2, 0.4, 0.8, 1.6, 2.4, 3.2],
  screens: {
    sm: '640px',
    md: '768px',
    lg: '1024px',
    xl: '1280px',
    xxl: '1536px',
  },
};

export default theme;
