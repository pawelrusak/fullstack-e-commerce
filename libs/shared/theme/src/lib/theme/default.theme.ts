import {
  color,
  fontFamily,
  fontWeight,
  space,
  screens,
  component,
} from './tokens';

import type { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  color,
  fontFamily,
  fontWeight,
  space,
  screens,
  component,
};
