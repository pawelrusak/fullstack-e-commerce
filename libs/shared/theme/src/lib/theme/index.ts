'use client';

export { default as theme } from './default.theme';
