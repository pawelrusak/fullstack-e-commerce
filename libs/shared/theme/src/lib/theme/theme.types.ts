import type {
  AppSpace,
  AppColor,
  AppFontFamily,
  AppFontWeight,
  AppScreens,
  AppComponent,
} from './tokens';

export type AppSpacesTheme = {
  space: AppSpace;
};

export type AppColorTheme = {
  color: AppColor;
};

export type AppFontFamilyTheme = {
  fontFamily: AppFontFamily;
};

export type AppFontWeightTheme = {
  fontWeight: AppFontWeight;
};

export type AppScreensTheme = {
  screens: AppScreens;
};

export type AppComponentTheme = {
  component: AppComponent;
};

export type AppTheme = AppColorTheme &
  AppFontFamilyTheme &
  AppFontWeightTheme &
  AppSpacesTheme &
  AppScreensTheme &
  AppComponentTheme;
