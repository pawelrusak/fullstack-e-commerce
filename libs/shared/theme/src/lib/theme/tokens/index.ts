export { component } from './component.tokens';
export { color, fontFamily, fontWeight, screens, space } from './system.tokens';
