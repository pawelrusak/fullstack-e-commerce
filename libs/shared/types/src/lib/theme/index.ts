export type * from './theme';
export type {
  BrandComponent,
  ButtonComponent,
  ButtonComponentPalette,
} from './component';
