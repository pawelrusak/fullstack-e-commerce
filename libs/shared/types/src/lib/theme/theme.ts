import { Theme as StyledSystemTheme } from 'styled-system';

// TODO: Move this file to the theme library
export type AppSpacesTheme = {
  space: StyledSystemTheme['space'];
};

export type AppColorTheme = {
  color: {
    primary: string;
    secondary: string;
    text: string;
    background: string;
    // TODO: Rename this to "secondarySurface". "Surface" will be used for UI components' backgrounds, such as cards, modals, navbar, etc.
    secondBackground: string;
    border: string;
    error: string;
  };
};

export type AppFontFamily = {
  fontFamily: {
    /**
     * @deprecated Use "primary" instead
     */
    montserrat: string;
    /**
     * @deprecated Use "secondary" instead.
     */
    poppins: string;
    /**
     * For body text, forms, etc.
     *
     * @warning Use only for headings at level H5 or lower.
     */
    primary?: string;
    /**
     * For headings, titles, etc.
     */
    secondary?: string;
  };
};

export type AppFontWeight = {
  fontWeight: {
    regular: number;
    semiBold: number;
    bold: number;
  };
};

export type AppScreensTheme = {
  screens: {
    sm: string;
    md: string;
    lg: string;
    xl: string;
    xxl: string;
  };
};

export type AppTheme = AppColorTheme &
  AppFontFamily &
  AppFontWeight &
  AppSpacesTheme &
  AppScreensTheme;
