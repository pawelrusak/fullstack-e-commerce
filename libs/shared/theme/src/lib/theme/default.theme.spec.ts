import { defaultTheme } from './default.theme';

describe('Default Theme', () => {
  it('should have the correct structure and types', () => {
    expect(defaultTheme).toEqual({
      color: expect.any(Object),
      fontFamily: expect.any(Object),
      fontWeight: expect.any(Object),
      space: expect.any(Object),
      screens: expect.any(Object),
      component: expect.any(Object),
    });
  });

  it('should match the specified values', () => {
    expect(defaultTheme).toMatchSnapshot();
  });
});
