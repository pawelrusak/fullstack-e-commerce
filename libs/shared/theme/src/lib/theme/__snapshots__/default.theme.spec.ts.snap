// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Theme should match the specified values 1`] = `
{
  "color": {
    "background": "hsl(0, 0%, 100%)",
    "border": "hsl(0, 0%, 96%)",
    "error": "hsl(348, 83%, 47%)",
    "onBackground": "hsl(0, 0%, 19%)",
    "onBackgroundInteract": "hsl(7, 100%, 65%)",
    "onBackgroundLink": "hsl(7, 100%, 65%)",
    "onBackgroundLinkInteract": "hsl(7, 100%, 59%)",
    "onPrimary": "hsl(0, 0%, 100%)",
    "onPrimaryContainer": "hsl(7, 100%, 65%)",
    "onPrimaryDisabled": "hsl(0, 0%, 100%)",
    "onPrimaryOutline": "hsl(7, 100%, 65%)",
    "onPrimaryOutlineDisabled": "hsl(0, 0%, 80%)",
    "onPrimaryOutlineInteract": "hsl(0, 0%, 100%)",
    "onSecondary": "hsl(0, 0%, 100%)",
    "onSecondaryDisabled": "hsl(0, 0%, 100%)",
    "onSecondaryOutline": "hsl(0, 0%, 19%)",
    "onSecondaryOutlineDisabled": "hsl(0, 0%, 80%)",
    "onSecondaryOutlineInteract": "hsl(0, 0%, 100%)",
    "onSecondarySurface": "hsl(0, 0%, 19%)",
    "onSecondarySurfaceVariant": "hsl(7, 100%, 65%)",
    "onSurface": "hsl(0, 0%, 19%)",
    "onSurfaceInteract": "hsl(7, 100%, 65%)",
    "onSurfaceVariant": "hsl(7, 100%, 65%)",
    "primary": "hsl(7, 100%, 65%)",
    "primaryBorder": "hsl(7, 100%, 65%)",
    "primaryBorderDisabled": "hsl(7, 100%, 81%)",
    "primaryBorderInteract": "hsl(7, 100%, 59%)",
    "primaryContainer": "hsl(7, 100%, 96%)",
    "primaryDisabled": "hsl(7, 100%, 81%)",
    "primaryInteract": "hsl(7, 100%, 59%)",
    "primaryOutline": "transparent",
    "primaryOutlineBorder": "hsl(7, 100%, 65%)",
    "primaryOutlineBorderDisabled": "hsl(0, 0%, 80%)",
    "primaryOutlineBorderInteract": "hsl(7, 100%, 59%)",
    "primaryOutlineDisabled": "transparent",
    "primaryOutlineInteract": "hsl(7, 100%, 59%)",
    "secondBackground": "hsl(0, 0%, 96%)",
    "secondary": "hsl(0, 0%, 19%)",
    "secondaryBorder": "hsl(0, 0%, 19%)",
    "secondaryBorderDisabled": "hsl(0, 0%, 80%)",
    "secondaryBorderInteract": "hsl(0, 0%, 16%)",
    "secondaryDisabled": "hsl(0, 0%, 80%)",
    "secondaryInteract": "hsl(0, 0%, 16%)",
    "secondaryOutline": "transparent",
    "secondaryOutlineBorder": "hsl(0, 0%, 19%)",
    "secondaryOutlineBorderDisabled": "hsl(0, 0%, 80%)",
    "secondaryOutlineBorderInteract": "hsl(0, 0%, 16%)",
    "secondaryOutlineDisabled": "transparent",
    "secondaryOutlineInteract": "hsl(0, 0%, 16%)",
    "secondarySurface": "hsl(0, 0%, 96%)",
    "surface": "hsl(0, 0%, 100%)",
    "surfacePlaceholder": "hsl(0, 0%, 82%)",
    "tertiary": "hsl(46, 99%, 65%)",
    "text": "hsl(0, 0%, 19%)",
  },
  "component": {
    "brand": {
      "svgDot": {
        "primary": {
          "initial": {
            "fill": "hsl(7, 100%, 65%)",
          },
        },
        "secondary": {
          "initial": {
            "fill": "hsl(0, 0%, 19%)",
          },
        },
      },
      "svgText": {
        "default": {
          "initial": {
            "fill": "hsl(0, 0%, 19%)",
          },
        },
        "invert": {
          "initial": {
            "fill": "hsl(0, 0%, 100%)",
          },
        },
      },
    },
    "breadcrumb": {
      "link": {
        "_base": {
          "backgroundColor": "transparent",
          "fontFamily": "Montserrat, sans-serif",
          "fontSize": "1.6rem",
          "lineHeight": 1.5,
        },
        "default": {
          "initial": {
            "color": "hsl(0, 0%, 19%)",
          },
          "interact": {
            "color": "hsl(7, 100%, 65%)",
          },
        },
        "isCurrent": {
          "current": {
            "fontWeight": 700,
          },
          "initial": {
            "fontWeight": 400,
          },
        },
      },
      "root": {
        "_base": {
          "backgroundColor": "transparent",
        },
      },
    },
    "button": {
      "root": {
        "_base": {
          "fontFamily": "Montserrat, sans-serif",
          "fontWeight": 700,
        },
        "size-default": {
          "initial": {
            "fontSize": "2rem",
            "lineHeight": "3rem",
            "padding": "1.5rem 3.9rem",
          },
        },
        "size-large": {
          "initial": {
            "fontSize": "2rem",
            "lineHeight": "3rem",
            "padding": "2.3rem 6.300000000000001rem",
          },
        },
        "size-small": {
          "initial": {
            "fontSize": "1.6rem",
            "lineHeight": "2.4rem",
            "padding": "0.7000000000000001rem 1.9rem",
          },
        },
        "variant-outline-primary": {
          "disabled": {
            "backgroundColor": "transparent",
            "borderColor": "hsl(0, 0%, 80%)",
            "color": "hsl(0, 0%, 80%)",
          },
          "initial": {
            "backgroundColor": "transparent",
            "borderColor": "hsl(7, 100%, 65%)",
            "color": "hsl(7, 100%, 65%)",
          },
          "interact": {
            "backgroundColor": "hsl(7, 100%, 59%)",
            "borderColor": "hsl(7, 100%, 59%)",
            "color": "hsl(0, 0%, 100%)",
          },
        },
        "variant-outline-secondary": {
          "disabled": {
            "backgroundColor": "transparent",
            "borderColor": "hsl(0, 0%, 80%)",
            "color": "hsl(0, 0%, 80%)",
          },
          "initial": {
            "backgroundColor": "transparent",
            "borderColor": "hsl(0, 0%, 19%)",
            "color": "hsl(0, 0%, 19%)",
          },
          "interact": {
            "backgroundColor": "hsl(0, 0%, 16%)",
            "borderColor": "hsl(0, 0%, 16%)",
            "color": "hsl(0, 0%, 100%)",
          },
        },
        "variant-primary": {
          "disabled": {
            "backgroundColor": "hsl(7, 100%, 81%)",
            "borderColor": "hsl(7, 100%, 81%)",
            "color": "hsl(0, 0%, 100%)",
          },
          "initial": {
            "backgroundColor": "hsl(7, 100%, 65%)",
            "borderColor": "hsl(7, 100%, 65%)",
            "color": "hsl(0, 0%, 100%)",
          },
          "interact": {
            "backgroundColor": "hsl(7, 100%, 59%)",
            "borderColor": "hsl(7, 100%, 59%)",
            "color": "hsl(0, 0%, 100%)",
          },
        },
        "variant-secondary": {
          "disabled": {
            "backgroundColor": "hsl(0, 0%, 80%)",
            "borderColor": "hsl(0, 0%, 80%)",
            "color": "hsl(0, 0%, 100%)",
          },
          "initial": {
            "backgroundColor": "hsl(0, 0%, 19%)",
            "borderColor": "hsl(0, 0%, 19%)",
            "color": "hsl(0, 0%, 100%)",
          },
          "interact": {
            "backgroundColor": "hsl(0, 0%, 16%)",
            "borderColor": "hsl(0, 0%, 16%)",
            "color": "hsl(0, 0%, 100%)",
          },
        },
      },
    },
    "cardsSection": {
      "body": {
        "_base": {
          "backgroundColor": "transparent",
          "color": "hsl(0, 0%, 19%)",
        },
      },
      "readMoreLink": {
        "_base": {
          "fontFamily": "Montserrat, sans-serif",
          "fontWeight": 700,
        },
        "default": {
          "initial": {
            "color": "hsl(7, 100%, 65%)",
          },
          "interact": {
            "color": "hsl(7, 100%, 59%)",
          },
        },
      },
      "root": {
        "_base": {
          "backgroundColor": "transparent",
        },
      },
      "title": {
        "_base": {
          "color": "hsl(0, 0%, 19%)",
          "fontFamily": "Poppins, sans-serif",
          "fontWeight": 600,
        },
      },
    },
    "cart": {
      "tableBody": {
        "default": {
          "initial": {
            "borderColor": "transparent",
          },
          "interact": {
            "borderColor": "hsl(0, 0%, 80%)",
          },
        },
      },
      "tableHead": {
        "_base": {
          "backgroundColor": "hsl(0, 0%, 96%)",
          "color": "hsl(0, 0%, 19%)",
          "fontFamily": "Montserrat, sans-serif",
          "fontSize": "1.6rem",
          "fontWeight": 700,
          "lineHeight": 1.5,
          "textTransform": "uppercase",
        },
      },
      "tableProductCategory": {
        "_base": {
          "fontSize": "1.4rem",
          "lineHeight": 1.4285714285714286,
          "textTransform": "uppercase",
        },
      },
      "tableProductName": {
        "_base": {
          "fontFamily": "Poppins, sans-serif",
          "fontSize": "2.0rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
      },
      "tableProductThumbnail": {
        "_base": {
          "backgroundColor": "hsl(0, 0%, 82%)",
        },
      },
      "tableProductVariantDetails": {
        "_base": {
          "fontSize": "1.6rem",
          "fontWeight": 400,
          "lineHeight": 1.5,
        },
      },
      "tableProductVariantLabel": {
        "_base": {
          "fontSize": "1.6rem",
          "fontWeight": 700,
          "lineHeight": 1.5,
        },
      },
    },
    "checkbox": {
      "indicator": {
        "_base": {
          "color": "hsl(0, 0%, 100%)",
        },
      },
      "root": {
        "default": {
          "checked": {
            "backgroundColor": "hsl(7, 100%, 65%)",
            "borderColor": "hsl(7, 100%, 65%)",
          },
          "initial": {
            "backgroundColor": "transparent",
            "borderColor": "hsl(0, 0%, 19%)",
          },
          "interact": {
            "backgroundColor": "hsl(7, 100%, 59%)",
            "borderColor": "hsl(7, 100%, 59%)",
          },
        },
      },
    },
    "divider": {
      "root": {
        "_base": {
          "borderColor": "hsl(0, 0%, 80%)",
        },
      },
    },
    "field": {
      "controlInput": {
        "default": {
          "focus": {
            "borderColor": "hsl(7, 100%, 65%)",
            "boxShadow": "0 0 0 0.3rem rgba(255,97,77,0.6)",
            "color": undefined,
          },
          "initial": {
            "borderColor": "hsl(0, 0%, 80%)",
            "boxShadow": undefined,
            "color": "hsl(0, 0%, 19%)",
          },
          "invalid": {
            "borderColor": "hsl(348, 83%, 47%)",
            "boxShadow": undefined,
            "color": "hsl(348, 83%, 47%)",
          },
        },
      },
      "controlInputPlaceholder": {
        "_base": {
          "fontFamily": "Montserrat, sans-serif",
          "fontWeight": 400,
        },
        "default": {
          "initial": {
            "color": "hsl(0, 0%, 19%)",
          },
          "invalid": {
            "color": "hsl(348, 83%, 47%)",
          },
        },
      },
      "errorText": {
        "_base": {
          "color": "hsl(348, 83%, 47%)",
        },
      },
      "helpText": {
        "_base": {
          "fontSize": "1.6rem",
          "lineHeight": "2.4rem",
        },
      },
      "label": {
        "_base": {
          "fontWeight": 700,
        },
      },
      "labelAsterisk": {
        "_base": {
          "color": "hsl(7, 100%, 65%)",
        },
      },
      "root": {
        "default": {
          "initial": {
            "color": "hsl(0, 0%, 19%)",
          },
          "invalid": {
            "color": "hsl(348, 83%, 47%)",
          },
        },
      },
    },
    "footer": {
      "copyrightBar": {
        "_base": {
          "borderTopColor": "hsl(0, 0%, 80%)",
        },
      },
      "copyrightBarFormula": {
        "_base": {
          "color": "hsl(0, 0%, 19%)",
        },
      },
      "copyrightBarListItemLink": {
        "default": {
          "initial": {
            "color": "hsl(0, 0%, 19%)",
          },
          "interact": {
            "color": "hsl(7, 100%, 65%)",
          },
        },
      },
      "copyrightBarListItemSeparator": {
        "_base": {
          "backgroundColor": "hsl(0, 0%, 80%)",
        },
      },
      "navSectionListItemIcon": {
        "_base": {
          "color": "rgba(48,48,48,0.75)",
        },
      },
      "navSectionListItemLink": {
        "default": {
          "initial": {
            "color": "hsl(0, 0%, 19%)",
          },
          "interact": {
            "color": "hsl(7, 100%, 65%)",
          },
        },
      },
      "navSectionTitle": {
        "_base": {
          "color": "hsl(0, 0%, 19%)",
          "fontFamily": "Poppins, sans-serif",
          "fontWeight": 600,
        },
      },
      "root": {
        "_base": {
          "borderTopColor": "hsl(0, 0%, 80%)",
        },
      },
    },
  },
  "fontFamily": {
    "montserrat": "Montserrat, sans-serif",
    "poppins": "Poppins, sans-serif",
    "primary": "Montserrat, sans-serif",
    "secondary": "Poppins, sans-serif",
  },
  "fontSizes": {
    "body": "2.0rem",
    "h1": "5.6rem",
    "h2": "3.6rem",
    "h3": "2.4rem",
    "h4": "2.0rem",
    "h5": "2.0rem",
    "h6": "1.6rem",
    "h7": "1.6rem",
    "h8": "1.4rem",
  },
  "fontWeight": {
    "bold": 700,
    "regular": 400,
    "semiBold": 600,
  },
  "lineHeights": {
    "body": 1.5,
    "h1": 1.2142857142857142,
    "h2": 1.2777777777777777,
    "h3": 1.25,
    "h4": 1.5,
    "h5": 1.5,
    "h6": 1.5,
    "h7": 1.5,
    "h8": 1.4285714285714286,
  },
  "screens": {
    "lg": "1024px",
    "md": "768px",
    "sm": "640px",
    "xl": "1280px",
    "xxl": "1536px",
  },
  "space": {
    "0": "0",
    "1": "0.4rem",
    "10": "8rem",
    "11": "10rem",
    "2": "0.8rem",
    "3": "1.6rem",
    "4": "2rem",
    "5": "2.4rem",
    "6": "3.2rem",
    "7": "4rem",
    "8": "4.8rem",
    "9": "6.4rem",
  },
}
`;
